src/Microsoft.AspNetCore.Mvc.Core/Internal/FileResultExecutorBase.cs

```diff
@@ -2,25 +2,54 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Headers;
+using Microsoft.AspNetCore.Internal;
 using Microsoft.Extensions.Logging;
 using Microsoft.Net.Http.Headers;
 
 namespace Microsoft.AspNetCore.Mvc.Internal
 {
     public class FileResultExecutorBase
     {
+        private const string AcceptRangeHeaderValue = "bytes";
+
         public FileResultExecutorBase(ILogger logger)
         {
             Logger = logger;
         }
 
         protected ILogger Logger { get; }
 
-        protected void SetHeadersAndLog(ActionContext context, FileResult result)
+        protected (RangeItemHeaderValue range, long rangeLength)? SetHeadersAndLog(ActionContext context, FileResult result, long fileLength, DateTimeOffset? lastModified = null, EntityTagHeaderValue etag = null)
         {
             SetContentType(context, result);
             SetContentDispositionHeader(context, result);
```

Comment:
If precondition fails 412 or , these are null and 0 respectively, which means the range will not be served (the entire file will be). No other Range headers will be set either. @Tratcher do the `Content-Range` and `Content-Length` need to be set for these cases? It doesn't sound like it. https://tools.ietf.org/html/rfc7233#section-4.2